File: Chatter.Web/src/app/Models/chatUser.ts
import { Contact } from "./contact";

export interface ChatUser
{
    id: string;
    userName: string;
    lastName: string;
    firstName: string;
    patronymic: string;
    universityName: string;
    universityFaculty: string;
    joined: Date;
    lastActive: Date;
    contacts: Contact[];
}

File: Chatter.Web/Chatter.Web/Models/ChatUser.cs
﻿namespace Chatter.Web.Models
{
    public class ChatUser
    {
        public Guid ID { get; set; }

        public string LastName { get; set; }

        public string FirstName { get; set; }

        public string? Patronymic { get; set; }

        public DateTime JoinedUtc { get; set; }

        public DateTime LastActiveUtc { get; set; }

        public bool IsBlocked { get; set; }

        public DateTime? BlockedUntilUtc { get; set; }

        public string? UniversityName { get; set; }

        public string? UniversityFaculty { get; set; }

        public List<PrivateChat> Contacts { get; set; }
    }
}
